{
    "version": "https://jsonfeed.org/version/1",
    "title": "Security Advisory for Python packages hosted at PyPI.org",
    "home_page_url": "https://github.com/advisories?query=type%3Areviewed+ecosystem%3Apip",
    "feed_url": "https://azu.github.io/github-advisory-database-rss/pip.json",
    "description": "Security Advisory for Python packages hosted at PyPI.org on GitHub",
    "items": [
        {
            "content_html": "<h3 id=\"impact\">Impact</h3>\n<p>The implementation of tf.reshape op in TensorFlow is vulnerable to a denial of service via CHECK-failure (assertion failure) caused by overflowing the number of elements in a tensor:</p>\n<pre><code class=\"language-python\">import tensorflow as tf\n\ntf.reshape(tensor=[[1]],shape=tf.constant([0 for i in range(255)], dtype=tf.int64))\n</code></pre>\n<p>This is another instance of <a href=\"https://github.com/tensorflow/tensorflow/blob/master/tensorflow/security/advisory/tfsa-2021-198.md\">TFSA-2021-198</a> (<a href=\"https://github.com/advisories/GHSA-prcg-wp5q-rv7p\">https://github.com/advisories/GHSA-prcg-wp5q-rv7p</a>).</p>\n<h3 id=\"patches\">Patches</h3>\n<p>We have patched the issue in GitHub commit <a href=\"https://github.com/tensorflow/tensorflow/commit/61f0f9b94df8c0411f0ad0ecc2fec2d3f3c33555\">61f0f9b94df8c0411f0ad0ecc2fec2d3f3c33555</a>.</p>\n<p>The fix will be included in TensorFlow 2.10.0. We will also cherrypick this commit on TensorFlow 2.9.1, TensorFlow 2.8.1, and TensorFlow 2.7.2, as these are also affected and still in supported range.</p>\n<h3 id=\"for-more-information\">For more information</h3>\n<p>Please consult <a href=\"https://github.com/tensorflow/tensorflow/blob/master/SECURITY.md\">our security guide</a> for more information regarding the security model and how to contact us with issues and questions.</p>\n<h3 id=\"attribution\">Attribution</h3>\n<p>This vulnerability has been reported by Kang Hong Jin from Singapore Management University.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://github.com/tensorflow/tensorflow/security/advisories/GHSA-f4w6-h4f5-wx45\">https://github.com/tensorflow/tensorflow/security/advisories/GHSA-f4w6-h4f5-wx45</a></li>\n<li><a href=\"https://github.com/tensorflow/tensorflow/commit/61f0f9b94df8c0411f0ad0ecc2fec2d3f3c33555\">https://github.com/tensorflow/tensorflow/commit/61f0f9b94df8c0411f0ad0ecc2fec2d3f3c33555</a></li>\n<li><a href=\"https://github.com/tensorflow/tensorflow/releases/tag/v2.10.0\">https://github.com/tensorflow/tensorflow/releases/tag/v2.10.0</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-f4w6-h4f5-wx45\">https://github.com/advisories/GHSA-f4w6-h4f5-wx45</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-f4w6-h4f5-wx45",
            "title": "[tensorflow-gpu] TensorFlow vulnerable to `CHECK` failure in tf.reshape via overflows",
            "date_modified": "2022-09-16T20:56:04.000Z",
            "date_published": "2022-09-16T20:56:02.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-f4w6-h4f5-wx45"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<h3 id=\"impact\">Impact</h3>\n<p>The implementation of tf.reshape op in TensorFlow is vulnerable to a denial of service via CHECK-failure (assertion failure) caused by overflowing the number of elements in a tensor:</p>\n<pre><code class=\"language-python\">import tensorflow as tf\n\ntf.reshape(tensor=[[1]],shape=tf.constant([0 for i in range(255)], dtype=tf.int64))\n</code></pre>\n<p>This is another instance of <a href=\"https://github.com/tensorflow/tensorflow/blob/master/tensorflow/security/advisory/tfsa-2021-198.md\">TFSA-2021-198</a> (<a href=\"https://github.com/advisories/GHSA-prcg-wp5q-rv7p\">https://github.com/advisories/GHSA-prcg-wp5q-rv7p</a>).</p>\n<h3 id=\"patches\">Patches</h3>\n<p>We have patched the issue in GitHub commit <a href=\"https://github.com/tensorflow/tensorflow/commit/61f0f9b94df8c0411f0ad0ecc2fec2d3f3c33555\">61f0f9b94df8c0411f0ad0ecc2fec2d3f3c33555</a>.</p>\n<p>The fix will be included in TensorFlow 2.10.0. We will also cherrypick this commit on TensorFlow 2.9.1, TensorFlow 2.8.1, and TensorFlow 2.7.2, as these are also affected and still in supported range.</p>\n<h3 id=\"for-more-information\">For more information</h3>\n<p>Please consult <a href=\"https://github.com/tensorflow/tensorflow/blob/master/SECURITY.md\">our security guide</a> for more information regarding the security model and how to contact us with issues and questions.</p>\n<h3 id=\"attribution\">Attribution</h3>\n<p>This vulnerability has been reported by Kang Hong Jin from Singapore Management University.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://github.com/tensorflow/tensorflow/security/advisories/GHSA-f4w6-h4f5-wx45\">https://github.com/tensorflow/tensorflow/security/advisories/GHSA-f4w6-h4f5-wx45</a></li>\n<li><a href=\"https://github.com/tensorflow/tensorflow/commit/61f0f9b94df8c0411f0ad0ecc2fec2d3f3c33555\">https://github.com/tensorflow/tensorflow/commit/61f0f9b94df8c0411f0ad0ecc2fec2d3f3c33555</a></li>\n<li><a href=\"https://github.com/tensorflow/tensorflow/releases/tag/v2.10.0\">https://github.com/tensorflow/tensorflow/releases/tag/v2.10.0</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-f4w6-h4f5-wx45\">https://github.com/advisories/GHSA-f4w6-h4f5-wx45</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-f4w6-h4f5-wx45",
            "title": "[tensorflow-gpu] TensorFlow vulnerable to `CHECK` failure in tf.reshape via overflows",
            "date_modified": "2022-09-16T20:56:04.000Z",
            "date_published": "2022-09-16T20:56:02.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-f4w6-h4f5-wx45"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<h3 id=\"impact\">Impact</h3>\n<p>The implementation of tf.reshape op in TensorFlow is vulnerable to a denial of service via CHECK-failure (assertion failure) caused by overflowing the number of elements in a tensor:</p>\n<pre><code class=\"language-python\">import tensorflow as tf\n\ntf.reshape(tensor=[[1]],shape=tf.constant([0 for i in range(255)], dtype=tf.int64))\n</code></pre>\n<p>This is another instance of <a href=\"https://github.com/tensorflow/tensorflow/blob/master/tensorflow/security/advisory/tfsa-2021-198.md\">TFSA-2021-198</a> (<a href=\"https://github.com/advisories/GHSA-prcg-wp5q-rv7p\">https://github.com/advisories/GHSA-prcg-wp5q-rv7p</a>).</p>\n<h3 id=\"patches\">Patches</h3>\n<p>We have patched the issue in GitHub commit <a href=\"https://github.com/tensorflow/tensorflow/commit/61f0f9b94df8c0411f0ad0ecc2fec2d3f3c33555\">61f0f9b94df8c0411f0ad0ecc2fec2d3f3c33555</a>.</p>\n<p>The fix will be included in TensorFlow 2.10.0. We will also cherrypick this commit on TensorFlow 2.9.1, TensorFlow 2.8.1, and TensorFlow 2.7.2, as these are also affected and still in supported range.</p>\n<h3 id=\"for-more-information\">For more information</h3>\n<p>Please consult <a href=\"https://github.com/tensorflow/tensorflow/blob/master/SECURITY.md\">our security guide</a> for more information regarding the security model and how to contact us with issues and questions.</p>\n<h3 id=\"attribution\">Attribution</h3>\n<p>This vulnerability has been reported by Kang Hong Jin from Singapore Management University.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://github.com/tensorflow/tensorflow/security/advisories/GHSA-f4w6-h4f5-wx45\">https://github.com/tensorflow/tensorflow/security/advisories/GHSA-f4w6-h4f5-wx45</a></li>\n<li><a href=\"https://github.com/tensorflow/tensorflow/commit/61f0f9b94df8c0411f0ad0ecc2fec2d3f3c33555\">https://github.com/tensorflow/tensorflow/commit/61f0f9b94df8c0411f0ad0ecc2fec2d3f3c33555</a></li>\n<li><a href=\"https://github.com/tensorflow/tensorflow/releases/tag/v2.10.0\">https://github.com/tensorflow/tensorflow/releases/tag/v2.10.0</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-f4w6-h4f5-wx45\">https://github.com/advisories/GHSA-f4w6-h4f5-wx45</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-f4w6-h4f5-wx45",
            "title": "[tensorflow-gpu] TensorFlow vulnerable to `CHECK` failure in tf.reshape via overflows",
            "date_modified": "2022-09-16T20:56:04.000Z",
            "date_published": "2022-09-16T20:56:02.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-f4w6-h4f5-wx45"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<h3 id=\"impact\">Impact</h3>\n<p>The implementation of tf.reshape op in TensorFlow is vulnerable to a denial of service via CHECK-failure (assertion failure) caused by overflowing the number of elements in a tensor:</p>\n<pre><code class=\"language-python\">import tensorflow as tf\n\ntf.reshape(tensor=[[1]],shape=tf.constant([0 for i in range(255)], dtype=tf.int64))\n</code></pre>\n<p>This is another instance of <a href=\"https://github.com/tensorflow/tensorflow/blob/master/tensorflow/security/advisory/tfsa-2021-198.md\">TFSA-2021-198</a> (<a href=\"https://github.com/advisories/GHSA-prcg-wp5q-rv7p\">https://github.com/advisories/GHSA-prcg-wp5q-rv7p</a>).</p>\n<h3 id=\"patches\">Patches</h3>\n<p>We have patched the issue in GitHub commit <a href=\"https://github.com/tensorflow/tensorflow/commit/61f0f9b94df8c0411f0ad0ecc2fec2d3f3c33555\">61f0f9b94df8c0411f0ad0ecc2fec2d3f3c33555</a>.</p>\n<p>The fix will be included in TensorFlow 2.10.0. We will also cherrypick this commit on TensorFlow 2.9.1, TensorFlow 2.8.1, and TensorFlow 2.7.2, as these are also affected and still in supported range.</p>\n<h3 id=\"for-more-information\">For more information</h3>\n<p>Please consult <a href=\"https://github.com/tensorflow/tensorflow/blob/master/SECURITY.md\">our security guide</a> for more information regarding the security model and how to contact us with issues and questions.</p>\n<h3 id=\"attribution\">Attribution</h3>\n<p>This vulnerability has been reported by Kang Hong Jin from Singapore Management University.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://github.com/tensorflow/tensorflow/security/advisories/GHSA-f4w6-h4f5-wx45\">https://github.com/tensorflow/tensorflow/security/advisories/GHSA-f4w6-h4f5-wx45</a></li>\n<li><a href=\"https://github.com/tensorflow/tensorflow/commit/61f0f9b94df8c0411f0ad0ecc2fec2d3f3c33555\">https://github.com/tensorflow/tensorflow/commit/61f0f9b94df8c0411f0ad0ecc2fec2d3f3c33555</a></li>\n<li><a href=\"https://github.com/tensorflow/tensorflow/releases/tag/v2.10.0\">https://github.com/tensorflow/tensorflow/releases/tag/v2.10.0</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-f4w6-h4f5-wx45\">https://github.com/advisories/GHSA-f4w6-h4f5-wx45</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-f4w6-h4f5-wx45",
            "title": "[tensorflow-cpu] TensorFlow vulnerable to `CHECK` failure in tf.reshape via overflows",
            "date_modified": "2022-09-16T20:56:04.000Z",
            "date_published": "2022-09-16T20:56:02.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-f4w6-h4f5-wx45"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<h3 id=\"impact\">Impact</h3>\n<p>The implementation of tf.reshape op in TensorFlow is vulnerable to a denial of service via CHECK-failure (assertion failure) caused by overflowing the number of elements in a tensor:</p>\n<pre><code class=\"language-python\">import tensorflow as tf\n\ntf.reshape(tensor=[[1]],shape=tf.constant([0 for i in range(255)], dtype=tf.int64))\n</code></pre>\n<p>This is another instance of <a href=\"https://github.com/tensorflow/tensorflow/blob/master/tensorflow/security/advisory/tfsa-2021-198.md\">TFSA-2021-198</a> (<a href=\"https://github.com/advisories/GHSA-prcg-wp5q-rv7p\">https://github.com/advisories/GHSA-prcg-wp5q-rv7p</a>).</p>\n<h3 id=\"patches\">Patches</h3>\n<p>We have patched the issue in GitHub commit <a href=\"https://github.com/tensorflow/tensorflow/commit/61f0f9b94df8c0411f0ad0ecc2fec2d3f3c33555\">61f0f9b94df8c0411f0ad0ecc2fec2d3f3c33555</a>.</p>\n<p>The fix will be included in TensorFlow 2.10.0. We will also cherrypick this commit on TensorFlow 2.9.1, TensorFlow 2.8.1, and TensorFlow 2.7.2, as these are also affected and still in supported range.</p>\n<h3 id=\"for-more-information\">For more information</h3>\n<p>Please consult <a href=\"https://github.com/tensorflow/tensorflow/blob/master/SECURITY.md\">our security guide</a> for more information regarding the security model and how to contact us with issues and questions.</p>\n<h3 id=\"attribution\">Attribution</h3>\n<p>This vulnerability has been reported by Kang Hong Jin from Singapore Management University.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://github.com/tensorflow/tensorflow/security/advisories/GHSA-f4w6-h4f5-wx45\">https://github.com/tensorflow/tensorflow/security/advisories/GHSA-f4w6-h4f5-wx45</a></li>\n<li><a href=\"https://github.com/tensorflow/tensorflow/commit/61f0f9b94df8c0411f0ad0ecc2fec2d3f3c33555\">https://github.com/tensorflow/tensorflow/commit/61f0f9b94df8c0411f0ad0ecc2fec2d3f3c33555</a></li>\n<li><a href=\"https://github.com/tensorflow/tensorflow/releases/tag/v2.10.0\">https://github.com/tensorflow/tensorflow/releases/tag/v2.10.0</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-f4w6-h4f5-wx45\">https://github.com/advisories/GHSA-f4w6-h4f5-wx45</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-f4w6-h4f5-wx45",
            "title": "[tensorflow-cpu] TensorFlow vulnerable to `CHECK` failure in tf.reshape via overflows",
            "date_modified": "2022-09-16T20:56:04.000Z",
            "date_published": "2022-09-16T20:56:02.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-f4w6-h4f5-wx45"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<h3 id=\"impact\">Impact</h3>\n<p>The implementation of tf.reshape op in TensorFlow is vulnerable to a denial of service via CHECK-failure (assertion failure) caused by overflowing the number of elements in a tensor:</p>\n<pre><code class=\"language-python\">import tensorflow as tf\n\ntf.reshape(tensor=[[1]],shape=tf.constant([0 for i in range(255)], dtype=tf.int64))\n</code></pre>\n<p>This is another instance of <a href=\"https://github.com/tensorflow/tensorflow/blob/master/tensorflow/security/advisory/tfsa-2021-198.md\">TFSA-2021-198</a> (<a href=\"https://github.com/advisories/GHSA-prcg-wp5q-rv7p\">https://github.com/advisories/GHSA-prcg-wp5q-rv7p</a>).</p>\n<h3 id=\"patches\">Patches</h3>\n<p>We have patched the issue in GitHub commit <a href=\"https://github.com/tensorflow/tensorflow/commit/61f0f9b94df8c0411f0ad0ecc2fec2d3f3c33555\">61f0f9b94df8c0411f0ad0ecc2fec2d3f3c33555</a>.</p>\n<p>The fix will be included in TensorFlow 2.10.0. We will also cherrypick this commit on TensorFlow 2.9.1, TensorFlow 2.8.1, and TensorFlow 2.7.2, as these are also affected and still in supported range.</p>\n<h3 id=\"for-more-information\">For more information</h3>\n<p>Please consult <a href=\"https://github.com/tensorflow/tensorflow/blob/master/SECURITY.md\">our security guide</a> for more information regarding the security model and how to contact us with issues and questions.</p>\n<h3 id=\"attribution\">Attribution</h3>\n<p>This vulnerability has been reported by Kang Hong Jin from Singapore Management University.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://github.com/tensorflow/tensorflow/security/advisories/GHSA-f4w6-h4f5-wx45\">https://github.com/tensorflow/tensorflow/security/advisories/GHSA-f4w6-h4f5-wx45</a></li>\n<li><a href=\"https://github.com/tensorflow/tensorflow/commit/61f0f9b94df8c0411f0ad0ecc2fec2d3f3c33555\">https://github.com/tensorflow/tensorflow/commit/61f0f9b94df8c0411f0ad0ecc2fec2d3f3c33555</a></li>\n<li><a href=\"https://github.com/tensorflow/tensorflow/releases/tag/v2.10.0\">https://github.com/tensorflow/tensorflow/releases/tag/v2.10.0</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-f4w6-h4f5-wx45\">https://github.com/advisories/GHSA-f4w6-h4f5-wx45</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-f4w6-h4f5-wx45",
            "title": "[tensorflow-cpu] TensorFlow vulnerable to `CHECK` failure in tf.reshape via overflows",
            "date_modified": "2022-09-16T20:56:04.000Z",
            "date_published": "2022-09-16T20:56:02.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-f4w6-h4f5-wx45"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<h3 id=\"impact\">Impact</h3>\n<p>The implementation of tf.reshape op in TensorFlow is vulnerable to a denial of service via CHECK-failure (assertion failure) caused by overflowing the number of elements in a tensor:</p>\n<pre><code class=\"language-python\">import tensorflow as tf\n\ntf.reshape(tensor=[[1]],shape=tf.constant([0 for i in range(255)], dtype=tf.int64))\n</code></pre>\n<p>This is another instance of <a href=\"https://github.com/tensorflow/tensorflow/blob/master/tensorflow/security/advisory/tfsa-2021-198.md\">TFSA-2021-198</a> (<a href=\"https://github.com/advisories/GHSA-prcg-wp5q-rv7p\">https://github.com/advisories/GHSA-prcg-wp5q-rv7p</a>).</p>\n<h3 id=\"patches\">Patches</h3>\n<p>We have patched the issue in GitHub commit <a href=\"https://github.com/tensorflow/tensorflow/commit/61f0f9b94df8c0411f0ad0ecc2fec2d3f3c33555\">61f0f9b94df8c0411f0ad0ecc2fec2d3f3c33555</a>.</p>\n<p>The fix will be included in TensorFlow 2.10.0. We will also cherrypick this commit on TensorFlow 2.9.1, TensorFlow 2.8.1, and TensorFlow 2.7.2, as these are also affected and still in supported range.</p>\n<h3 id=\"for-more-information\">For more information</h3>\n<p>Please consult <a href=\"https://github.com/tensorflow/tensorflow/blob/master/SECURITY.md\">our security guide</a> for more information regarding the security model and how to contact us with issues and questions.</p>\n<h3 id=\"attribution\">Attribution</h3>\n<p>This vulnerability has been reported by Kang Hong Jin from Singapore Management University.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://github.com/tensorflow/tensorflow/security/advisories/GHSA-f4w6-h4f5-wx45\">https://github.com/tensorflow/tensorflow/security/advisories/GHSA-f4w6-h4f5-wx45</a></li>\n<li><a href=\"https://github.com/tensorflow/tensorflow/commit/61f0f9b94df8c0411f0ad0ecc2fec2d3f3c33555\">https://github.com/tensorflow/tensorflow/commit/61f0f9b94df8c0411f0ad0ecc2fec2d3f3c33555</a></li>\n<li><a href=\"https://github.com/tensorflow/tensorflow/releases/tag/v2.10.0\">https://github.com/tensorflow/tensorflow/releases/tag/v2.10.0</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-f4w6-h4f5-wx45\">https://github.com/advisories/GHSA-f4w6-h4f5-wx45</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-f4w6-h4f5-wx45",
            "title": "[tensorflow] TensorFlow vulnerable to `CHECK` failure in tf.reshape via overflows",
            "date_modified": "2022-09-16T20:56:04.000Z",
            "date_published": "2022-09-16T20:56:02.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-f4w6-h4f5-wx45"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<h3 id=\"impact\">Impact</h3>\n<p>The implementation of tf.reshape op in TensorFlow is vulnerable to a denial of service via CHECK-failure (assertion failure) caused by overflowing the number of elements in a tensor:</p>\n<pre><code class=\"language-python\">import tensorflow as tf\n\ntf.reshape(tensor=[[1]],shape=tf.constant([0 for i in range(255)], dtype=tf.int64))\n</code></pre>\n<p>This is another instance of <a href=\"https://github.com/tensorflow/tensorflow/blob/master/tensorflow/security/advisory/tfsa-2021-198.md\">TFSA-2021-198</a> (<a href=\"https://github.com/advisories/GHSA-prcg-wp5q-rv7p\">https://github.com/advisories/GHSA-prcg-wp5q-rv7p</a>).</p>\n<h3 id=\"patches\">Patches</h3>\n<p>We have patched the issue in GitHub commit <a href=\"https://github.com/tensorflow/tensorflow/commit/61f0f9b94df8c0411f0ad0ecc2fec2d3f3c33555\">61f0f9b94df8c0411f0ad0ecc2fec2d3f3c33555</a>.</p>\n<p>The fix will be included in TensorFlow 2.10.0. We will also cherrypick this commit on TensorFlow 2.9.1, TensorFlow 2.8.1, and TensorFlow 2.7.2, as these are also affected and still in supported range.</p>\n<h3 id=\"for-more-information\">For more information</h3>\n<p>Please consult <a href=\"https://github.com/tensorflow/tensorflow/blob/master/SECURITY.md\">our security guide</a> for more information regarding the security model and how to contact us with issues and questions.</p>\n<h3 id=\"attribution\">Attribution</h3>\n<p>This vulnerability has been reported by Kang Hong Jin from Singapore Management University.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://github.com/tensorflow/tensorflow/security/advisories/GHSA-f4w6-h4f5-wx45\">https://github.com/tensorflow/tensorflow/security/advisories/GHSA-f4w6-h4f5-wx45</a></li>\n<li><a href=\"https://github.com/tensorflow/tensorflow/commit/61f0f9b94df8c0411f0ad0ecc2fec2d3f3c33555\">https://github.com/tensorflow/tensorflow/commit/61f0f9b94df8c0411f0ad0ecc2fec2d3f3c33555</a></li>\n<li><a href=\"https://github.com/tensorflow/tensorflow/releases/tag/v2.10.0\">https://github.com/tensorflow/tensorflow/releases/tag/v2.10.0</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-f4w6-h4f5-wx45\">https://github.com/advisories/GHSA-f4w6-h4f5-wx45</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-f4w6-h4f5-wx45",
            "title": "[tensorflow] TensorFlow vulnerable to `CHECK` failure in tf.reshape via overflows",
            "date_modified": "2022-09-16T20:56:04.000Z",
            "date_published": "2022-09-16T20:56:02.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-f4w6-h4f5-wx45"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<h3 id=\"impact\">Impact</h3>\n<p>The implementation of tf.reshape op in TensorFlow is vulnerable to a denial of service via CHECK-failure (assertion failure) caused by overflowing the number of elements in a tensor:</p>\n<pre><code class=\"language-python\">import tensorflow as tf\n\ntf.reshape(tensor=[[1]],shape=tf.constant([0 for i in range(255)], dtype=tf.int64))\n</code></pre>\n<p>This is another instance of <a href=\"https://github.com/tensorflow/tensorflow/blob/master/tensorflow/security/advisory/tfsa-2021-198.md\">TFSA-2021-198</a> (<a href=\"https://github.com/advisories/GHSA-prcg-wp5q-rv7p\">https://github.com/advisories/GHSA-prcg-wp5q-rv7p</a>).</p>\n<h3 id=\"patches\">Patches</h3>\n<p>We have patched the issue in GitHub commit <a href=\"https://github.com/tensorflow/tensorflow/commit/61f0f9b94df8c0411f0ad0ecc2fec2d3f3c33555\">61f0f9b94df8c0411f0ad0ecc2fec2d3f3c33555</a>.</p>\n<p>The fix will be included in TensorFlow 2.10.0. We will also cherrypick this commit on TensorFlow 2.9.1, TensorFlow 2.8.1, and TensorFlow 2.7.2, as these are also affected and still in supported range.</p>\n<h3 id=\"for-more-information\">For more information</h3>\n<p>Please consult <a href=\"https://github.com/tensorflow/tensorflow/blob/master/SECURITY.md\">our security guide</a> for more information regarding the security model and how to contact us with issues and questions.</p>\n<h3 id=\"attribution\">Attribution</h3>\n<p>This vulnerability has been reported by Kang Hong Jin from Singapore Management University.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://github.com/tensorflow/tensorflow/security/advisories/GHSA-f4w6-h4f5-wx45\">https://github.com/tensorflow/tensorflow/security/advisories/GHSA-f4w6-h4f5-wx45</a></li>\n<li><a href=\"https://github.com/tensorflow/tensorflow/commit/61f0f9b94df8c0411f0ad0ecc2fec2d3f3c33555\">https://github.com/tensorflow/tensorflow/commit/61f0f9b94df8c0411f0ad0ecc2fec2d3f3c33555</a></li>\n<li><a href=\"https://github.com/tensorflow/tensorflow/releases/tag/v2.10.0\">https://github.com/tensorflow/tensorflow/releases/tag/v2.10.0</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-f4w6-h4f5-wx45\">https://github.com/advisories/GHSA-f4w6-h4f5-wx45</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-f4w6-h4f5-wx45",
            "title": "[tensorflow] TensorFlow vulnerable to `CHECK` failure in tf.reshape via overflows",
            "date_modified": "2022-09-16T20:56:04.000Z",
            "date_published": "2022-09-16T20:56:02.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-f4w6-h4f5-wx45"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<h3 id=\"impact\">Impact</h3>\n<p>The implementation of SobolSampleOp is vulnerable to a denial of service via CHECK-failure (assertion failure) caused by assuming <code>input(0)</code>, <code>input(1)</code>, and <code>input(2)</code> to be scalar.</p>\n<pre><code class=\"language-python\">import tensorflow as tf\ntf.raw_ops.SobolSample(dim=tf.constant([1,0]), num_results=tf.constant([1]), skip=tf.constant([1]))\n</code></pre>\n<h3 id=\"patches\">Patches</h3>\n<p>We have patched the issue in GitHub commit <a href=\"https://github.com/tensorflow/tensorflow/commit/c65c67f88ad770662e8f191269a907bf2b94b1bf\">c65c67f88ad770662e8f191269a907bf2b94b1bf</a>.</p>\n<p>The fix will be included in TensorFlow 2.10.0. We will also cherrypick this commit on TensorFlow 2.9.1, TensorFlow 2.8.1, and TensorFlow 2.7.2, as these are also affected and still in supported range.</p>\n<h3 id=\"for-more-information\">For more information</h3>\n<p>Please consult <a href=\"https://github.com/tensorflow/tensorflow/blob/master/SECURITY.md\">our security guide</a> for more information regarding the security model and how to contact us with issues and questions.</p>\n<h3 id=\"attribution\">Attribution</h3>\n<p>This vulnerability has been reported by:</p>\n<ul>\n<li>Kang Hong Jin from Singapore Management University</li>\n<li>Neophytos Christou, Secure Systems Labs, Brown University</li>\n<li>刘力源, Information System &amp; Security and Countermeasures Experiments Center, Beijing Institute of Technology</li>\n</ul>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://github.com/tensorflow/tensorflow/security/advisories/GHSA-97p7-w86h-vcf9\">https://github.com/tensorflow/tensorflow/security/advisories/GHSA-97p7-w86h-vcf9</a></li>\n<li><a href=\"https://github.com/tensorflow/tensorflow/commit/c65c67f88ad770662e8f191269a907bf2b94b1bf\">https://github.com/tensorflow/tensorflow/commit/c65c67f88ad770662e8f191269a907bf2b94b1bf</a></li>\n<li><a href=\"https://github.com/tensorflow/tensorflow/releases/tag/v2.10.0\">https://github.com/tensorflow/tensorflow/releases/tag/v2.10.0</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-97p7-w86h-vcf9\">https://github.com/advisories/GHSA-97p7-w86h-vcf9</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-97p7-w86h-vcf9",
            "title": "[tensorflow-gpu] TensorFlow vulnerable to `CHECK` failure in `SobolSample` via missing validation",
            "date_modified": "2022-09-16T20:51:55.000Z",
            "date_published": "2022-09-16T20:51:55.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-97p7-w86h-vcf9"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<h3 id=\"impact\">Impact</h3>\n<p>The implementation of SobolSampleOp is vulnerable to a denial of service via CHECK-failure (assertion failure) caused by assuming <code>input(0)</code>, <code>input(1)</code>, and <code>input(2)</code> to be scalar.</p>\n<pre><code class=\"language-python\">import tensorflow as tf\ntf.raw_ops.SobolSample(dim=tf.constant([1,0]), num_results=tf.constant([1]), skip=tf.constant([1]))\n</code></pre>\n<h3 id=\"patches\">Patches</h3>\n<p>We have patched the issue in GitHub commit <a href=\"https://github.com/tensorflow/tensorflow/commit/c65c67f88ad770662e8f191269a907bf2b94b1bf\">c65c67f88ad770662e8f191269a907bf2b94b1bf</a>.</p>\n<p>The fix will be included in TensorFlow 2.10.0. We will also cherrypick this commit on TensorFlow 2.9.1, TensorFlow 2.8.1, and TensorFlow 2.7.2, as these are also affected and still in supported range.</p>\n<h3 id=\"for-more-information\">For more information</h3>\n<p>Please consult <a href=\"https://github.com/tensorflow/tensorflow/blob/master/SECURITY.md\">our security guide</a> for more information regarding the security model and how to contact us with issues and questions.</p>\n<h3 id=\"attribution\">Attribution</h3>\n<p>This vulnerability has been reported by:</p>\n<ul>\n<li>Kang Hong Jin from Singapore Management University</li>\n<li>Neophytos Christou, Secure Systems Labs, Brown University</li>\n<li>刘力源, Information System &amp; Security and Countermeasures Experiments Center, Beijing Institute of Technology</li>\n</ul>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://github.com/tensorflow/tensorflow/security/advisories/GHSA-97p7-w86h-vcf9\">https://github.com/tensorflow/tensorflow/security/advisories/GHSA-97p7-w86h-vcf9</a></li>\n<li><a href=\"https://github.com/tensorflow/tensorflow/commit/c65c67f88ad770662e8f191269a907bf2b94b1bf\">https://github.com/tensorflow/tensorflow/commit/c65c67f88ad770662e8f191269a907bf2b94b1bf</a></li>\n<li><a href=\"https://github.com/tensorflow/tensorflow/releases/tag/v2.10.0\">https://github.com/tensorflow/tensorflow/releases/tag/v2.10.0</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-97p7-w86h-vcf9\">https://github.com/advisories/GHSA-97p7-w86h-vcf9</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-97p7-w86h-vcf9",
            "title": "[tensorflow-gpu] TensorFlow vulnerable to `CHECK` failure in `SobolSample` via missing validation",
            "date_modified": "2022-09-16T20:51:55.000Z",
            "date_published": "2022-09-16T20:51:55.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-97p7-w86h-vcf9"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<h3 id=\"impact\">Impact</h3>\n<p>The implementation of SobolSampleOp is vulnerable to a denial of service via CHECK-failure (assertion failure) caused by assuming <code>input(0)</code>, <code>input(1)</code>, and <code>input(2)</code> to be scalar.</p>\n<pre><code class=\"language-python\">import tensorflow as tf\ntf.raw_ops.SobolSample(dim=tf.constant([1,0]), num_results=tf.constant([1]), skip=tf.constant([1]))\n</code></pre>\n<h3 id=\"patches\">Patches</h3>\n<p>We have patched the issue in GitHub commit <a href=\"https://github.com/tensorflow/tensorflow/commit/c65c67f88ad770662e8f191269a907bf2b94b1bf\">c65c67f88ad770662e8f191269a907bf2b94b1bf</a>.</p>\n<p>The fix will be included in TensorFlow 2.10.0. We will also cherrypick this commit on TensorFlow 2.9.1, TensorFlow 2.8.1, and TensorFlow 2.7.2, as these are also affected and still in supported range.</p>\n<h3 id=\"for-more-information\">For more information</h3>\n<p>Please consult <a href=\"https://github.com/tensorflow/tensorflow/blob/master/SECURITY.md\">our security guide</a> for more information regarding the security model and how to contact us with issues and questions.</p>\n<h3 id=\"attribution\">Attribution</h3>\n<p>This vulnerability has been reported by:</p>\n<ul>\n<li>Kang Hong Jin from Singapore Management University</li>\n<li>Neophytos Christou, Secure Systems Labs, Brown University</li>\n<li>刘力源, Information System &amp; Security and Countermeasures Experiments Center, Beijing Institute of Technology</li>\n</ul>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://github.com/tensorflow/tensorflow/security/advisories/GHSA-97p7-w86h-vcf9\">https://github.com/tensorflow/tensorflow/security/advisories/GHSA-97p7-w86h-vcf9</a></li>\n<li><a href=\"https://github.com/tensorflow/tensorflow/commit/c65c67f88ad770662e8f191269a907bf2b94b1bf\">https://github.com/tensorflow/tensorflow/commit/c65c67f88ad770662e8f191269a907bf2b94b1bf</a></li>\n<li><a href=\"https://github.com/tensorflow/tensorflow/releases/tag/v2.10.0\">https://github.com/tensorflow/tensorflow/releases/tag/v2.10.0</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-97p7-w86h-vcf9\">https://github.com/advisories/GHSA-97p7-w86h-vcf9</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-97p7-w86h-vcf9",
            "title": "[tensorflow-gpu] TensorFlow vulnerable to `CHECK` failure in `SobolSample` via missing validation",
            "date_modified": "2022-09-16T20:51:55.000Z",
            "date_published": "2022-09-16T20:51:55.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-97p7-w86h-vcf9"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<h3 id=\"impact\">Impact</h3>\n<p>The implementation of SobolSampleOp is vulnerable to a denial of service via CHECK-failure (assertion failure) caused by assuming <code>input(0)</code>, <code>input(1)</code>, and <code>input(2)</code> to be scalar.</p>\n<pre><code class=\"language-python\">import tensorflow as tf\ntf.raw_ops.SobolSample(dim=tf.constant([1,0]), num_results=tf.constant([1]), skip=tf.constant([1]))\n</code></pre>\n<h3 id=\"patches\">Patches</h3>\n<p>We have patched the issue in GitHub commit <a href=\"https://github.com/tensorflow/tensorflow/commit/c65c67f88ad770662e8f191269a907bf2b94b1bf\">c65c67f88ad770662e8f191269a907bf2b94b1bf</a>.</p>\n<p>The fix will be included in TensorFlow 2.10.0. We will also cherrypick this commit on TensorFlow 2.9.1, TensorFlow 2.8.1, and TensorFlow 2.7.2, as these are also affected and still in supported range.</p>\n<h3 id=\"for-more-information\">For more information</h3>\n<p>Please consult <a href=\"https://github.com/tensorflow/tensorflow/blob/master/SECURITY.md\">our security guide</a> for more information regarding the security model and how to contact us with issues and questions.</p>\n<h3 id=\"attribution\">Attribution</h3>\n<p>This vulnerability has been reported by:</p>\n<ul>\n<li>Kang Hong Jin from Singapore Management University</li>\n<li>Neophytos Christou, Secure Systems Labs, Brown University</li>\n<li>刘力源, Information System &amp; Security and Countermeasures Experiments Center, Beijing Institute of Technology</li>\n</ul>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://github.com/tensorflow/tensorflow/security/advisories/GHSA-97p7-w86h-vcf9\">https://github.com/tensorflow/tensorflow/security/advisories/GHSA-97p7-w86h-vcf9</a></li>\n<li><a href=\"https://github.com/tensorflow/tensorflow/commit/c65c67f88ad770662e8f191269a907bf2b94b1bf\">https://github.com/tensorflow/tensorflow/commit/c65c67f88ad770662e8f191269a907bf2b94b1bf</a></li>\n<li><a href=\"https://github.com/tensorflow/tensorflow/releases/tag/v2.10.0\">https://github.com/tensorflow/tensorflow/releases/tag/v2.10.0</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-97p7-w86h-vcf9\">https://github.com/advisories/GHSA-97p7-w86h-vcf9</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-97p7-w86h-vcf9",
            "title": "[tensorflow-cpu] TensorFlow vulnerable to `CHECK` failure in `SobolSample` via missing validation",
            "date_modified": "2022-09-16T20:51:55.000Z",
            "date_published": "2022-09-16T20:51:55.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-97p7-w86h-vcf9"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<h3 id=\"impact\">Impact</h3>\n<p>The implementation of SobolSampleOp is vulnerable to a denial of service via CHECK-failure (assertion failure) caused by assuming <code>input(0)</code>, <code>input(1)</code>, and <code>input(2)</code> to be scalar.</p>\n<pre><code class=\"language-python\">import tensorflow as tf\ntf.raw_ops.SobolSample(dim=tf.constant([1,0]), num_results=tf.constant([1]), skip=tf.constant([1]))\n</code></pre>\n<h3 id=\"patches\">Patches</h3>\n<p>We have patched the issue in GitHub commit <a href=\"https://github.com/tensorflow/tensorflow/commit/c65c67f88ad770662e8f191269a907bf2b94b1bf\">c65c67f88ad770662e8f191269a907bf2b94b1bf</a>.</p>\n<p>The fix will be included in TensorFlow 2.10.0. We will also cherrypick this commit on TensorFlow 2.9.1, TensorFlow 2.8.1, and TensorFlow 2.7.2, as these are also affected and still in supported range.</p>\n<h3 id=\"for-more-information\">For more information</h3>\n<p>Please consult <a href=\"https://github.com/tensorflow/tensorflow/blob/master/SECURITY.md\">our security guide</a> for more information regarding the security model and how to contact us with issues and questions.</p>\n<h3 id=\"attribution\">Attribution</h3>\n<p>This vulnerability has been reported by:</p>\n<ul>\n<li>Kang Hong Jin from Singapore Management University</li>\n<li>Neophytos Christou, Secure Systems Labs, Brown University</li>\n<li>刘力源, Information System &amp; Security and Countermeasures Experiments Center, Beijing Institute of Technology</li>\n</ul>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://github.com/tensorflow/tensorflow/security/advisories/GHSA-97p7-w86h-vcf9\">https://github.com/tensorflow/tensorflow/security/advisories/GHSA-97p7-w86h-vcf9</a></li>\n<li><a href=\"https://github.com/tensorflow/tensorflow/commit/c65c67f88ad770662e8f191269a907bf2b94b1bf\">https://github.com/tensorflow/tensorflow/commit/c65c67f88ad770662e8f191269a907bf2b94b1bf</a></li>\n<li><a href=\"https://github.com/tensorflow/tensorflow/releases/tag/v2.10.0\">https://github.com/tensorflow/tensorflow/releases/tag/v2.10.0</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-97p7-w86h-vcf9\">https://github.com/advisories/GHSA-97p7-w86h-vcf9</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-97p7-w86h-vcf9",
            "title": "[tensorflow-cpu] TensorFlow vulnerable to `CHECK` failure in `SobolSample` via missing validation",
            "date_modified": "2022-09-16T20:51:55.000Z",
            "date_published": "2022-09-16T20:51:55.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-97p7-w86h-vcf9"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<h3 id=\"impact\">Impact</h3>\n<p>The implementation of SobolSampleOp is vulnerable to a denial of service via CHECK-failure (assertion failure) caused by assuming <code>input(0)</code>, <code>input(1)</code>, and <code>input(2)</code> to be scalar.</p>\n<pre><code class=\"language-python\">import tensorflow as tf\ntf.raw_ops.SobolSample(dim=tf.constant([1,0]), num_results=tf.constant([1]), skip=tf.constant([1]))\n</code></pre>\n<h3 id=\"patches\">Patches</h3>\n<p>We have patched the issue in GitHub commit <a href=\"https://github.com/tensorflow/tensorflow/commit/c65c67f88ad770662e8f191269a907bf2b94b1bf\">c65c67f88ad770662e8f191269a907bf2b94b1bf</a>.</p>\n<p>The fix will be included in TensorFlow 2.10.0. We will also cherrypick this commit on TensorFlow 2.9.1, TensorFlow 2.8.1, and TensorFlow 2.7.2, as these are also affected and still in supported range.</p>\n<h3 id=\"for-more-information\">For more information</h3>\n<p>Please consult <a href=\"https://github.com/tensorflow/tensorflow/blob/master/SECURITY.md\">our security guide</a> for more information regarding the security model and how to contact us with issues and questions.</p>\n<h3 id=\"attribution\">Attribution</h3>\n<p>This vulnerability has been reported by:</p>\n<ul>\n<li>Kang Hong Jin from Singapore Management University</li>\n<li>Neophytos Christou, Secure Systems Labs, Brown University</li>\n<li>刘力源, Information System &amp; Security and Countermeasures Experiments Center, Beijing Institute of Technology</li>\n</ul>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://github.com/tensorflow/tensorflow/security/advisories/GHSA-97p7-w86h-vcf9\">https://github.com/tensorflow/tensorflow/security/advisories/GHSA-97p7-w86h-vcf9</a></li>\n<li><a href=\"https://github.com/tensorflow/tensorflow/commit/c65c67f88ad770662e8f191269a907bf2b94b1bf\">https://github.com/tensorflow/tensorflow/commit/c65c67f88ad770662e8f191269a907bf2b94b1bf</a></li>\n<li><a href=\"https://github.com/tensorflow/tensorflow/releases/tag/v2.10.0\">https://github.com/tensorflow/tensorflow/releases/tag/v2.10.0</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-97p7-w86h-vcf9\">https://github.com/advisories/GHSA-97p7-w86h-vcf9</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-97p7-w86h-vcf9",
            "title": "[tensorflow-cpu] TensorFlow vulnerable to `CHECK` failure in `SobolSample` via missing validation",
            "date_modified": "2022-09-16T20:51:55.000Z",
            "date_published": "2022-09-16T20:51:55.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-97p7-w86h-vcf9"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<h3 id=\"impact\">Impact</h3>\n<p>The implementation of SobolSampleOp is vulnerable to a denial of service via CHECK-failure (assertion failure) caused by assuming <code>input(0)</code>, <code>input(1)</code>, and <code>input(2)</code> to be scalar.</p>\n<pre><code class=\"language-python\">import tensorflow as tf\ntf.raw_ops.SobolSample(dim=tf.constant([1,0]), num_results=tf.constant([1]), skip=tf.constant([1]))\n</code></pre>\n<h3 id=\"patches\">Patches</h3>\n<p>We have patched the issue in GitHub commit <a href=\"https://github.com/tensorflow/tensorflow/commit/c65c67f88ad770662e8f191269a907bf2b94b1bf\">c65c67f88ad770662e8f191269a907bf2b94b1bf</a>.</p>\n<p>The fix will be included in TensorFlow 2.10.0. We will also cherrypick this commit on TensorFlow 2.9.1, TensorFlow 2.8.1, and TensorFlow 2.7.2, as these are also affected and still in supported range.</p>\n<h3 id=\"for-more-information\">For more information</h3>\n<p>Please consult <a href=\"https://github.com/tensorflow/tensorflow/blob/master/SECURITY.md\">our security guide</a> for more information regarding the security model and how to contact us with issues and questions.</p>\n<h3 id=\"attribution\">Attribution</h3>\n<p>This vulnerability has been reported by:</p>\n<ul>\n<li>Kang Hong Jin from Singapore Management University</li>\n<li>Neophytos Christou, Secure Systems Labs, Brown University</li>\n<li>刘力源, Information System &amp; Security and Countermeasures Experiments Center, Beijing Institute of Technology</li>\n</ul>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://github.com/tensorflow/tensorflow/security/advisories/GHSA-97p7-w86h-vcf9\">https://github.com/tensorflow/tensorflow/security/advisories/GHSA-97p7-w86h-vcf9</a></li>\n<li><a href=\"https://github.com/tensorflow/tensorflow/commit/c65c67f88ad770662e8f191269a907bf2b94b1bf\">https://github.com/tensorflow/tensorflow/commit/c65c67f88ad770662e8f191269a907bf2b94b1bf</a></li>\n<li><a href=\"https://github.com/tensorflow/tensorflow/releases/tag/v2.10.0\">https://github.com/tensorflow/tensorflow/releases/tag/v2.10.0</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-97p7-w86h-vcf9\">https://github.com/advisories/GHSA-97p7-w86h-vcf9</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-97p7-w86h-vcf9",
            "title": "[tensorflow] TensorFlow vulnerable to `CHECK` failure in `SobolSample` via missing validation",
            "date_modified": "2022-09-16T20:51:55.000Z",
            "date_published": "2022-09-16T20:51:55.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-97p7-w86h-vcf9"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<h3 id=\"impact\">Impact</h3>\n<p>The implementation of SobolSampleOp is vulnerable to a denial of service via CHECK-failure (assertion failure) caused by assuming <code>input(0)</code>, <code>input(1)</code>, and <code>input(2)</code> to be scalar.</p>\n<pre><code class=\"language-python\">import tensorflow as tf\ntf.raw_ops.SobolSample(dim=tf.constant([1,0]), num_results=tf.constant([1]), skip=tf.constant([1]))\n</code></pre>\n<h3 id=\"patches\">Patches</h3>\n<p>We have patched the issue in GitHub commit <a href=\"https://github.com/tensorflow/tensorflow/commit/c65c67f88ad770662e8f191269a907bf2b94b1bf\">c65c67f88ad770662e8f191269a907bf2b94b1bf</a>.</p>\n<p>The fix will be included in TensorFlow 2.10.0. We will also cherrypick this commit on TensorFlow 2.9.1, TensorFlow 2.8.1, and TensorFlow 2.7.2, as these are also affected and still in supported range.</p>\n<h3 id=\"for-more-information\">For more information</h3>\n<p>Please consult <a href=\"https://github.com/tensorflow/tensorflow/blob/master/SECURITY.md\">our security guide</a> for more information regarding the security model and how to contact us with issues and questions.</p>\n<h3 id=\"attribution\">Attribution</h3>\n<p>This vulnerability has been reported by:</p>\n<ul>\n<li>Kang Hong Jin from Singapore Management University</li>\n<li>Neophytos Christou, Secure Systems Labs, Brown University</li>\n<li>刘力源, Information System &amp; Security and Countermeasures Experiments Center, Beijing Institute of Technology</li>\n</ul>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://github.com/tensorflow/tensorflow/security/advisories/GHSA-97p7-w86h-vcf9\">https://github.com/tensorflow/tensorflow/security/advisories/GHSA-97p7-w86h-vcf9</a></li>\n<li><a href=\"https://github.com/tensorflow/tensorflow/commit/c65c67f88ad770662e8f191269a907bf2b94b1bf\">https://github.com/tensorflow/tensorflow/commit/c65c67f88ad770662e8f191269a907bf2b94b1bf</a></li>\n<li><a href=\"https://github.com/tensorflow/tensorflow/releases/tag/v2.10.0\">https://github.com/tensorflow/tensorflow/releases/tag/v2.10.0</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-97p7-w86h-vcf9\">https://github.com/advisories/GHSA-97p7-w86h-vcf9</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-97p7-w86h-vcf9",
            "title": "[tensorflow] TensorFlow vulnerable to `CHECK` failure in `SobolSample` via missing validation",
            "date_modified": "2022-09-16T20:51:55.000Z",
            "date_published": "2022-09-16T20:51:55.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-97p7-w86h-vcf9"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<h3 id=\"impact\">Impact</h3>\n<p>The implementation of SobolSampleOp is vulnerable to a denial of service via CHECK-failure (assertion failure) caused by assuming <code>input(0)</code>, <code>input(1)</code>, and <code>input(2)</code> to be scalar.</p>\n<pre><code class=\"language-python\">import tensorflow as tf\ntf.raw_ops.SobolSample(dim=tf.constant([1,0]), num_results=tf.constant([1]), skip=tf.constant([1]))\n</code></pre>\n<h3 id=\"patches\">Patches</h3>\n<p>We have patched the issue in GitHub commit <a href=\"https://github.com/tensorflow/tensorflow/commit/c65c67f88ad770662e8f191269a907bf2b94b1bf\">c65c67f88ad770662e8f191269a907bf2b94b1bf</a>.</p>\n<p>The fix will be included in TensorFlow 2.10.0. We will also cherrypick this commit on TensorFlow 2.9.1, TensorFlow 2.8.1, and TensorFlow 2.7.2, as these are also affected and still in supported range.</p>\n<h3 id=\"for-more-information\">For more information</h3>\n<p>Please consult <a href=\"https://github.com/tensorflow/tensorflow/blob/master/SECURITY.md\">our security guide</a> for more information regarding the security model and how to contact us with issues and questions.</p>\n<h3 id=\"attribution\">Attribution</h3>\n<p>This vulnerability has been reported by:</p>\n<ul>\n<li>Kang Hong Jin from Singapore Management University</li>\n<li>Neophytos Christou, Secure Systems Labs, Brown University</li>\n<li>刘力源, Information System &amp; Security and Countermeasures Experiments Center, Beijing Institute of Technology</li>\n</ul>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://github.com/tensorflow/tensorflow/security/advisories/GHSA-97p7-w86h-vcf9\">https://github.com/tensorflow/tensorflow/security/advisories/GHSA-97p7-w86h-vcf9</a></li>\n<li><a href=\"https://github.com/tensorflow/tensorflow/commit/c65c67f88ad770662e8f191269a907bf2b94b1bf\">https://github.com/tensorflow/tensorflow/commit/c65c67f88ad770662e8f191269a907bf2b94b1bf</a></li>\n<li><a href=\"https://github.com/tensorflow/tensorflow/releases/tag/v2.10.0\">https://github.com/tensorflow/tensorflow/releases/tag/v2.10.0</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-97p7-w86h-vcf9\">https://github.com/advisories/GHSA-97p7-w86h-vcf9</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-97p7-w86h-vcf9",
            "title": "[tensorflow] TensorFlow vulnerable to `CHECK` failure in `SobolSample` via missing validation",
            "date_modified": "2022-09-16T20:51:55.000Z",
            "date_published": "2022-09-16T20:51:55.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-97p7-w86h-vcf9"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>indy-node is the server portion of Hyperledger Indy, a distributed ledger purpose-built for decentralized identity.</p>\n<h3 id=\"impact\">Impact</h3>\n<p>An attacker can max out the number of client connections allowed by the ledger that was deployed using guidance provided in the indy-node repository, leaving the ledger unable to be used for its intended purpose.</p>\n<p>The ledger content will not be impacted by the attack, and the ledger will resume servicing valid client requests after the attack.</p>\n<h3 id=\"mitigations\">Mitigations</h3>\n<p>This attack exploits the trade-off between resilience and availability. Any protection against abusive client connections will also prevent the network being accessed by certain legitimate users. As a result, validator nodes must tune their firewall rules to ensure the right trade-off for their network's expected users. The guidance previously provided enabled a low-cost DDoS attack.</p>\n<p>The <a href=\"https://github.com/hyperledger/indy-node/blob/main/docs/source/setup-iptables.md\">guidance to network operators for the use of firewall rules</a> in the deployment of Indy networks has been modified to better protect against denial of service attacks by increasing the cost and complexity in mounting such attacks.</p>\n<p>The mitigation for this vulnerability is not in the Hyperledger Indy code per se, but rather in the individual deployments of Indy. The mitigations should be applied to all deployments of Indy, and are not related to a particular release.</p>\n<h3 id=\"acknowledgements\">Acknowledgements</h3>\n<p>Thank you to Mirko Mollik at <a href=\"https://trusterts.de\">TrustCerts.de</a> for finding and responsibly disclosing this issue.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://github.com/hyperledger/indy-node/security/advisories/GHSA-x996-7qh9-7ff7\">https://github.com/hyperledger/indy-node/security/advisories/GHSA-x996-7qh9-7ff7</a></li>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2022-31006\">https://nvd.nist.gov/vuln/detail/CVE-2022-31006</a></li>\n<li><a href=\"https://github.com/hyperledger/indy-node/commit/53a2a1bf1a26cb8ba710fd6adc8bcf275186a4b3\">https://github.com/hyperledger/indy-node/commit/53a2a1bf1a26cb8ba710fd6adc8bcf275186a4b3</a></li>\n<li><a href=\"https://github.com/hyperledger/indy-node/commits/v1.13.2-rc2\">https://github.com/hyperledger/indy-node/commits/v1.13.2-rc2</a></li>\n<li><a href=\"https://github.com/pypa/advisory-database/tree/main/vulns/indy-node/PYSEC-2022-270.yaml\">https://github.com/pypa/advisory-database/tree/main/vulns/indy-node/PYSEC-2022-270.yaml</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-x996-7qh9-7ff7\">https://github.com/advisories/GHSA-x996-7qh9-7ff7</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-x996-7qh9-7ff7",
            "title": "[indy-node] Hyperledger indy-node vulnerable to denial of service",
            "date_modified": "2022-09-16T20:28:40.000Z",
            "date_published": "2022-09-16T20:28:39.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-x996-7qh9-7ff7"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>A flaw was found in python-oslo-utils. Due to improper parsing, passwords with a double quote ( \" ) in them cause incorrect masking in debug logs, causing any part of the password after the double quote to be plaintext.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2022-0718\">https://nvd.nist.gov/vuln/detail/CVE-2022-0718</a></li>\n<li><a href=\"https://access.redhat.com/security/cve/CVE-2022-0718\">https://access.redhat.com/security/cve/CVE-2022-0718</a></li>\n<li><a href=\"https://bugs.launchpad.net/oslo.utils/+bug/1949623\">https://bugs.launchpad.net/oslo.utils/+bug/1949623</a></li>\n<li><a href=\"https://bugzilla.redhat.com/show_bug.cgi?id=2056850\">https://bugzilla.redhat.com/show_bug.cgi?id=2056850</a></li>\n<li><a href=\"https://opendev.org/openstack/oslo.utils/commit/6e17ae1f7959c64dfd20a5f67edf422e702426aa\">https://opendev.org/openstack/oslo.utils/commit/6e17ae1f7959c64dfd20a5f67edf422e702426aa</a></li>\n<li><a href=\"https://security-tracker.debian.org/tracker/CVE-2022-0718\">https://security-tracker.debian.org/tracker/CVE-2022-0718</a></li>\n<li><a href=\"https://github.com/pypa/advisory-database/tree/main/vulns/oslo-utils/PYSEC-2022-258.yaml\">https://github.com/pypa/advisory-database/tree/main/vulns/oslo-utils/PYSEC-2022-258.yaml</a></li>\n<li><a href=\"https://lists.debian.org/debian-lts-announce/2022/09/msg00015.html\">https://lists.debian.org/debian-lts-announce/2022/09/msg00015.html</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-wmqq-r32m-87c5\">https://github.com/advisories/GHSA-wmqq-r32m-87c5</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-wmqq-r32m-87c5",
            "title": "[oslo-utils] python-oslo-utils has improper password parsing",
            "date_modified": "2022-09-16T19:32:40.000Z",
            "date_published": "2022-08-29T20:06:50.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-wmqq-r32m-87c5"
            },
            "tags": [
                "severity"
            ]
        }
    ]
}